### ISSUE TITLE
Savon#call Raises Wasabi::Resolver::HTTPError 

### ISSUE THREAD

**Issue by sshaw** [savonrb/savon#654]
This implementation detail should not be leaked. A `Savon::Error` [should be raised] (http://savonrb.com/version2/errors.html) instead. 

Here's the stacktrace:

```
"/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/wasabi-3.3.0/lib/wasabi/resolver.rb:45:in `load_from_remote'",
 "/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/wasabi-3.3.0/lib/wasabi/resolver.rb:33:in `resolve'",
 "/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/wasabi-3.3.0/lib/wasabi/document.rb:142:in `xml'",
 "/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/wasabi-3.3.0/lib/wasabi/document.rb:160:in `parse'",
 "/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/wasabi-3.3.0/lib/wasabi/document.rb:147:in `parser'",
 "/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/wasabi-3.3.0/lib/wasabi/document.rb:64:in `soap_actions'",
 "/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/savon-2.8.0/lib/savon/operation.rb:21:in `ensure_exists!'",
 "/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/savon-2.8.0/lib/savon/operation.rb:14:in `create'",
 "/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/savon-2.8.0/lib/savon/client.rb:32:in `operation'",
 "/Users/sshaw/.rvm/gems/ruby-2.1.2@provident_client/gems/savon-2.8.0/lib/savon/client.rb:36:in `call'"
```

**Comment by tjarratt** [comment]
I agree 100% -- a `Savon::Error` would be much nicer. Thank you @sshaw for providing that stack trace.

I'm not sure if I'll be able to fix this anytime soon -- would you be interested in submitting a pull request for this?

**Comment by sshaw** [comment]
I've made to commits to my fork: 
https://github.com/sshaw/savon/commit/ae6128847cc336c1b18b576e71f37a1d2e112a93
https://github.com/sshaw/savon/commit/ae6128847cc336c1b18b576e71f37a1d2e112a93

but was having trouble getting [the spec](https://github.com/sshaw/savon/blob/wasabi_exception/spec/savon/operation_spec.rb#L48-51) to do what I think the spec infrastructure should. Some help here would be appreciated. Esentially I expect the `:timeout` endpoint to be called on the local server but it's not.  

As for the fix: The `Wasabi::Document` is the source of this but its docs don't explicitly state  what exceptions it raises, it looks to be only `HTTPError` but Further investigation revels that HTTPI suffers from a similar issue in that it raises a [`SocketError`](http://www.ruby-doc.org/stdlib-2.2.0/libdoc/socket/rdoc/SocketError.html). Is this intentional? I had code to deal with this case but removed it. 

I'm raising an `HTTPError` as it *is* an HTTP error and is consistent with the `#call` method's behavior. While included in the exception here `Wasabi::Resolver::HTTPError#response` is optional so this change is somewhat questionable.

**Comment by tjarratt** [comment]
The operation specs are rather interesting -- the WSDL is read from a fixture on disk, but requests to the server can be provided. So, in this case no exception is raised because verifying that the operation exists does not result in fetching anything from the server.

Effectively, you're hitting this code in `wasabi/resolver.rb` and hitting the XML case, which doesn't actually make any HTTP requests.

`wasabi/lib/wasabi/resolver.rb`
```ruby
    def resolve
      raise ArgumentError, "Unable to resolve: #{document.inspect}" unless document

      case document
        when URL then load_from_remote # you would expect to hit this path to raise an error
        when XML then document  # but this path is taken because the document provided is an XML file
        else          load_from_disc
      end
```

I feel like stubbing out `Wasabi::Document#soap_actions` to raise is a valid strategy here. Normally, I like integration tests, but this feels like more of a unit test to me.

**Comment by sshaw** [comment]
> So, in this case no exception is raised because verifying that the operation exists 
> does not result in fetching anything from the server

Ah yes, was too wrapped in my schema that needed to hit the new to resolve refs in the WSDL. 

Pull request here: #655

**Comment by tjarratt** [comment]
Closed by pull request #655. Thanks again @sshaw!
